virt/kvm/introspection/kvmi.c

```diff
@@ -464,6 +459,15 @@ alloc_kvmi(struct kvm *kvm, const struct kvm_introspection_hook *hook)
 		}
 	}
 
+	kvmi->access_tree = krealloc(kvmi->access_tree,
+				     kvm->arch.mmu_root_hpa_altviews_count *
+				     sizeof(struct radix_tree_root),
+				     GFP_KERNEL_ACCOUNT);
```

Comment:
`kvmi` is created in this function and `krealloc` doesn't make sense here, but it should be used when `mmu_root_hpa_altviews_count` is changed. Do I miss something?

Reply:
Thanks for noticing this. Should be fixed now : )